ci-automation/vendor-testing/aws.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+# Copyright (c) 2022 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+# Test execution script for the AWS vendor image.
+# This script is supposed to run in the mantle container.
+
+work_dir="$1"; shift
+arch="$1"; shift
+vernum="$1"; shift
+tapfile="$1"; shift
+
+# $@ now contains tests / test patterns to run
+
+source ci-automation/ci_automation_common.sh
+source sdk_lib/sdk_container_common.sh
+
+mkdir -p "${work_dir}"
+cd "${work_dir}"
+
+AWS_BOARD="${arch}-usr"
+AWS_CHANNEL="$(get_git_channel)"
+if [[ "${AWS_CHANNEL}" = 'developer' ]]; then
+    AWS_CHANNEL='alpha'
+fi
+AWS_IMAGE_NAME="ci-${vernum}"
+AWS_INSTANCE_TYPE_VAR="AWS_${arch}_INSTANCE_TYPE"
+AWS_INSTANCE_TYPE="${!AWS_INSTANCE_TYPE_VAR}"
+MORE_INSTANCE_TYPES_VAR="AWS_${arch}_MORE_INSTANCE_TYPES"
+MORE_INSTANCE_TYPES=( ${!MORE_INSTANCE_TYPES_VAR} )
+
+if [[ -z "${AWS_OFFER}" ]]; then
+    AWS_OFFER='basic'
+fi
+
+case "${AWS_OFFER}" in
+    basic)
+        AWS_OEM_SUFFIX=''
+        ;;
+    pro)
+        AWS_OEM_SUFFIX='_pro'
+        ;;
+    *)
+        echo "1..1" > "${tapfile}"
+        echo "not ok - all AWS tests" >> "${tapfile}"
+        echo "  ---" >> "${tapfile}"
+        echo "  ERROR: Invalid offer '${AWS_OFFER}', should be 'basic' or 'pro'." | tee -a "${tapfile}"
+        echo "  ..." >> "${tapfile}"
+        exit 1
+        ;;
+esac
+
+testscript="$(basename "$0")"
+
+if [[ "${AWS_AMI_ID}" == "" ]]; then
```

Comment:
Hm, then I'll need to see if it's possible to avoid downloading and uploading the image, but instead just pass some image URL to ore to create the image. Also, as I said above - I couldn't find any ami_vmdk image to use (unless I build one myself in CI).

Reply:
We can keep it then for local testing :)
Yeah, the image isn't built yet in the new pipeline, one would need to use it from a release https://stable.release.flatcar-linux.net/amd64-usr/current/flatcar_production_ami_vmdk_image.vmdk.bz2